src/main/java/com/bootcamp/demo/controller/AccountController.java

```diff
@@ -26,6 +30,37 @@ public AccountController(Service<Account> service) {
         }
     }
 
+    @PutMapping("/editAccount")
+    ResponseEntity<Account> update(@RequestBody AccountDetails accountDetails) throws ControllerException {
+        try {
+            Account searchedAccount = service.findOne(accountDetails.getUsername());
+
+            if (searchedAccount != null) {
+                Account account = new Account(
+                        accountDetails.getName(),
+                        accountDetails.getEmail(),
+                        accountDetails.getUsername(),
+                        searchedAccount.getPassword(),
+                        accountDetails.getPhoneNumber(),
+                        accountDetails.getAddress(),
+                        accountDetails.getDateOfBirth()
+                );
+
+                if (!Objects.equals(searchedAccount.getEmail(), accountDetails.getEmail())) {
+                    // TODO: Find a solution to return the error message: "Email cannot be changed!"
```

Comment:
You can if you change the update method from the service to return a string. Otherwise you can return a raw ResponseEntity. Ideally we wouldn't have to worry about this if we don't allow the user to change the email from the UI in the first place.